plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt

```diff
@@ -154,6 +154,10 @@ class CodeWhispererCodeScanManager(val project: Project) {
             codeScanJob = launchCodeScanCoroutine(CodeWhispererConstants.CodeAnalysisScope.PROJECT)
         } else {
             if (CodeWhispererExplorerActionManager.getInstance().isAutoEnabledForCodeScan() and !isUserBuilderId(project)) {
+                // cancel if a file scan is running.
+                if (!isCodeScanInProgress() && this::codeScanJob.isInitialized && codeScanJob.isActive) {
+                    codeScanJob.cancel()
+                }
```

Comment:
off topic, just curious, why don't we merge these 2 lines 
```
private val isCodeScanInProgress = AtomicBoolean(false)
private lateinit var codeScanJob: Job
```

to 

```
private var codeScanJob: AtomicReference<Job?>
```

and say `isCodeScanInProgress` if the value is non-null

Reply:
The `isCodeScanInProgress` is only tracking project scan(i should probably change its name). Using codeScanJob to track that could get confusing. As we don't want to cancel project scan for file scans since project scans always take precedence.